### ISSUE TITLE
Publish on open-vsx.org

### ISSUE THREAD

**Issue by bubba** [JustusAdam/language-haskell#171]
Hi, we're trying to upload the vscode-haskell extension to open-vsx.org here: https://github.com/haskell/vscode-haskell/issues/250, but we have a dependency in our package.json on this marvellous extension. Would it be possible to upload this to open-vsx.org as well? Since it seems like extensions can't be uploaded until all their dependencies are also uploaded

**Comment by bubba** [comment]
We've been following this guide to publish the extension: https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#1-create-an-access-token

**Comment by jankeromnes** [comment]
@bubba I would suggest keeping this feature request open.

As a quickfix, we are happy to add extensions to our auto-publishing CI in https://github.com/open-vsx/publish-extensions, but it's quite basic and bears the risk of not publishing exactly when/how/what is desired. So in general it's always better if maintainers take control of when/how/what they publish to OpenVSX. (And it's usually as easy as adding the single command `npx ovsx publish` to an existing Microsoft Marketplace publishing script.)

**Comment by marcin-rzeznicki** [comment]
`3.4.0` seems not to have been published to [OpenVSX](https://open-vsx.org/extension/justusadam/language-haskell) still - pinging

**Comment by filiptronicek** [comment]
> `3.4.0` seems not to have been published to [OpenVSX](https://open-vsx.org/extension/justusadam/language-haskell) still - pinging

That is correct indeed, the publishing script in the https://github.com/open-vsx/publish-extensions repository is [having some issues](https://github.com/open-vsx/publish-extensions/runs/5543731498?check_suite_focus=true#step:8:5971). In the meantime, I have created [a doc](https://github.com/open-vsx/publish-extensions/blob/master/docs/direct_publish_setup.md) describing the process on publishing, and I believe it should be pretty straight forward. I would love to help out!

cc @JustusAdam

**Comment by marcin-rzeznicki** [comment]
Thanks @filiptronicek :rocket:

**Comment by JustusAdam** [comment]
> @bubba I would suggest keeping this feature request open.
> 
> As a quickfix, we are happy to add extensions to our auto-publishing CI in https://github.com/open-vsx/publish-extensions, but it's quite basic and bears the risk of not publishing exactly when/how/what is desired. So in general it's always better if maintainers take control of when/how/what they publish to OpenVSX. (And it's usually as easy as adding the single command `npx ovsx publish` to an existing Microsoft Marketplace publishing script.)

Since there were some other issues with my not-yet-automated publishing recently I might spend some time to do that and I'd be happy to add the mentioned publishing command as well. Are there access control stuff I'd have to take care of to get that set up now?

**Comment by filiptronicek** [comment]
> I'd be happy to add the mentioned publishing command as well.

Great!

> Are there access control stuff I'd have to take care of to get that set up now?

This boils down to creating an account on open-vsx.org and getting a personal access token. This is described in the step one of the aforementioned doc, linking to [Publishing Extensions
](https://github.com/eclipse/openvsx/wiki/Publishing-Extensions), where only the steps 1-4 are important for now. The rest happens in the CI pipeline itself.

**Comment by re-xyr** [comment]
There is currently a `v3.6.0` on Open VSX, but it is broken (#212). How could we fix this?
